p2p/security/noise/transport.go

```diff
@@ -65,3 +82,50 @@ func (t *Transport) WithSessionOptions(opts ...SessionOption) (sec.SecureTranspo
 	}
 	return st, nil
 }
+
+func MatchMuxers(initiatorMuxers, responderMuxers []string) string {
+	respMuxerMap := make(map[string]struct{}, len(responderMuxers))
+	for _, muxer := range responderMuxers {
+		respMuxerMap[muxer] = struct{}{}
+	}
+
+	selectedMuxer := ""
+	for _, muxer := range initiatorMuxers {
+		if _, ok := respMuxerMap[muxer]; ok {
+			selectedMuxer = muxer
+			break
+		}
+	}
+	return selectedMuxer
+}
+
+type TransportEarlyDataHandler struct {
+	transport      *Transport
+	receivedMuxers []string
+}
+
+func NewTransportEDH(t *Transport) *TransportEarlyDataHandler {
```

Comment:
Have you considered reusing the `SessionTransport`?

Reply:
yes actually my first version of the implementation tried that approach of "withOptions". Then it becomes obvious that there are needs to do condition checks and customizations, my take is that the current approach is cleaner and will introduce less interference with webtransport code.